packages/core/src/routes/interaction/verifications/mfa-verification.ts

```diff
@@ -0,0 +1,86 @@
+import { InteractionEvent, MfaFactor, MfaPolicy } from '@logto/schemas';
+
+import RequestError from '#src/errors/RequestError/index.js';
+import type TenantContext from '#src/tenants/TenantContext.js';
+import assertThat from '#src/utils/assert-that.js';
+
+import { type WithInteractionDetailsContext } from '../middleware/koa-interaction-details.js';
+import { type WithInteractionSieContext } from '../middleware/koa-interaction-sie.js';
+import {
+  type VerifiedSignInInteractionResult,
+  type VerifiedInteractionResult,
+  type VerifiedRegisterInteractionResult,
+} from '../types/index.js';
+
+export const verifyBindMfa = async (
+  tenant: TenantContext,
+  interaction: VerifiedSignInInteractionResult | VerifiedRegisterInteractionResult
+): Promise<VerifiedInteractionResult> => {
+  const { bindMfa, event } = interaction;
+
+  if (!bindMfa || event !== InteractionEvent.SignIn) {
+    return interaction;
+  }
+
+  const { type } = bindMfa;
+
+  // There will be more types later
+  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+  if (type === MfaFactor.TOTP) {
+    const { accountId } = interaction;
+    const { mfaVerifications } = await tenant.queries.users.findUserById(accountId);
+
+    // A user can only bind one TOTP factor
+    assertThat(
+      mfaVerifications.every(({ type }) => type !== MfaFactor.TOTP),
+      new RequestError({
+        code: 'user.totp_already_in_use',
+        status: 422,
+      })
+    );
+  }
+
+  return interaction;
+};
+
+export const validateMandatoryBindMfa = async (
```

Comment:
can Logto user specify the type of mandatory MFA method? e.g. End-users have to bind a TOTP MFA, whether to bind a WebAuthn is optional. Can we support this case with current implementation?

Reply:
Yep, this will come when implementing another factor.